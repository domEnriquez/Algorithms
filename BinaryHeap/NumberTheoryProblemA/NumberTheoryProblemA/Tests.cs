using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NumberTheoryProblemA
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class Tests
    {
        StringBuilder fakeOutput;
        EquationSolution es;

        [SetUp]
        public void SetUp()
        {
            es = new EquationSolution();
            fakeOutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeOutput));
        }

        [Test]
        public void numLimitIs0()
        {
            es.FindSolution(0);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void numLimitIs1()
        {
            es.FindSolution(1);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void numLimitIs2()
        {
            es.FindSolution(2);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void numLimitis9K()
        {
            es.FindSolution(9000);

            Approvals.Verify(fakeOutput);
        }
    }
}
