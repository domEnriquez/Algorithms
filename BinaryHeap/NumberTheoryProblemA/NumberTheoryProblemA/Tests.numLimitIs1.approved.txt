0 = 0 + 2(0^2)
0 = 3(0^3) + 4(0^4)
3 = 1 + 2(1^2)
3 = 3(1^3) + 4(0^4)
